src/ICSharpCode.SharpZipLib/Zip/ZipExtraData.cs

```diff
@@ -474,6 +474,110 @@ public DateTime LastAccessTime
 		#endregion Instance Fields
 	}
 
+	/// <summary>
+	/// A TaggedData for handling WinzipAES extra data.
+	/// </summary>
+	/// <remarks>
+	/// See http://www.winzip.com/aes_info.htm for format documentation.
+	/// </remarks>
+	internal class WinZipAESTaggedData : ITaggedData
+	{
+		/// <summary>
+		/// The Version used by this entry.
+		/// </summary>
+		/// <remarks>
+		/// AE-1 iS 1. AE-2 is 2. With AE-2 no CRC is required and 0 is stored.
+		/// </remarks>
+		public enum VendorVersion : int
+		{
+			/// <summary>
+			/// Version AE-1.
+			/// </summary>
+			AE1 = 0x0001,
+
+			/// <summary>
+			/// Version AE-2.
+			/// </summary>
+			AE2 = 0x0002,
+		}
+
+		/// <summary>
+		/// Get the ID for this tagged data value.
+		/// </summary>
+		public ushort TagID => 0x9901;
+
+		/// <summary>
+		/// Set the data from the raw values provided.
+		/// </summary>
+		/// <param name="data">The raw data to extract values from.</param>
+		/// <param name="index">The index to start extracting values from.</param>
+		/// <param name="count">The number of bytes available.</param>
+		public void SetData(byte[] data, int index, int count)
+		{
+			using (MemoryStream ms = new MemoryStream(data, index, count, false))
+			using (ZipHelperStream helperStream = new ZipHelperStream(ms))
+			{
+				//
+				// Unpack AES extra data field see http://www.winzip.com/aes_info.htm
+				var length = helperStream.Length;         // Data size currently 7
+				if (length < 7)
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/piksel/SharpZipLib/pullRequest?prid=5745473)